File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/Penetrable.java
package org.tendiwa.plane.geometry.straightSkeleton;

import java.util.Optional;
import org.tendiwa.geometry.points.Point;
import org.tendiwa.geometry.segments.Segment;

interface Penetrable {
    void add(Point poll, Point poll1);

    Optional<Point> obtainIntersectionPoint(
        Segment intersected,
        Segment intersecting
    );

    double depth();
}

File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/Penetrable.kt
package org.tendiwa.plane.geometry.straightSkeleton

import org.tendiwa.geometry.points.Point
import org.tendiwa.geometry.segments.Segment
import java.util.*

internal interface Penetrable {
    fun add(poll: Point, poll1: Point)

    fun obtainIntersectionPoint(
        intersected: Segment,
        intersecting: Segment): Optional<Point>

    fun depth(): Double
}
